ygot/render.go

```diff
@@ -40,16 +40,33 @@ type path struct {
 	p []interface{}
 }
 
+// GNMINotificationsConfig specifies arguments determining how the
+// gNMI output should be created by ygot.
+type GNMINotificationsConfig struct {
+	// UsePathElem indicates that the PathElem message should be used
+	// to indicate path elements, using the elem field in the gNMI Path
+	// message. This path format was introduced in gNMI 0.4.0. If set
+	// to false, the path format used in the pre-0.4.0 format using
+	// a slice of strings as the path elements.
+	UsePathElem bool
+}
+
 // TogNMINotifications takes an input GoStruct and renders it to slice of
 // Notification messages, marked with the specified timestamp. If a parentPath
 // is used, it is used as a prefix path for the notifications returned.
-func TogNMINotifications(s GoStruct, ts int64, prefix []interface{}) ([]*gnmipb.Notification, error) {
+func TogNMINotifications(s GoStruct, ts int64, prefix []interface{}, cfg *GNMINotificationsConfig) ([]*gnmipb.Notification, error) {
```

Comment:
I'm a bit worried about simulating polymorphism with an interface and a bool here :-)
A cleaner way to achieve this would be to define a base type that can hold either type of path and some methods to emit or transform the path into something the helper functions are expecting (rather than passing the bool all the way down the chain).

Reply:
That's fair -- let me refactor. I was initially attempting to keep the public API unchanged (since there is code that uses this running elsewhere), but since there was a config public API change, then I think making this some form of typed prefix and within the config would probably be possible. This then allows cleaner handling of the paths as you say.